pkg/windows/calico.go

```diff
@@ -52,147 +53,226 @@ const (
+	c.CalicoCNICfg = &CNIConfig{NetworkName: "Calico", BindAddress: nodeConfig.AgentConfig.NodeIP}
+
+	platformType, err := getPlatformType()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	if err := getDefaultConfig(cfg, dataDir, nodeConfig); err != nil {
-		return nil, err
+	c.CalicoCNICfg.CalicoConfig = &CalicoConfig{
+		Name:                  "Calico",
+		Hostname:              nodeConfig.AgentConfig.NodeName,
+		NodeNameFile:          filepath.Join("c:\\", c.DataDir, "agent", CalicoNodeNameFileName),
+		KubeNetwork:           "Calico.*",
+		Mode:                  "vxlan",
+		ServiceCIDR:           nodeConfig.AgentConfig.ServiceCIDR.String(),
+		DNSServers:            nodeConfig.AgentConfig.ClusterDNS.String(),
+		DNSSearch:             "svc." + nodeConfig.AgentConfig.ClusterDomain,
+		DatastoreType:         "kubernetes",
+		NetworkingBackend:     "vxlan",
+		Platform:              platformType,
+		StartUpValidIPTimeout: 90,
+		LogDir:                "",
+		IP:                    "autodetect",
+		IPAutoDetectionMethod: "first-found",
+		Felix: FelixConfig{
+			Metadataaddr:    "none",
+			Vxlanvni:        "4096",
+			MacPrefix:       "0E-2A",
+			LogSeverityFile: "none",
+			LogSeveritySys:  "none",
+		},
+		CNI: CalicoCNIConfig{
+			BinDir:   nodeConfig.AgentConfig.CNIBinDir,
+			ConfDir:  nodeConfig.AgentConfig.CNIConfDir,
+			IpamType: "calico-ipam",
+			Version:  "0.3.1",
+		},
 	}
 
-	cfg.CalicoConfig.KubeConfig = calicoKubeConfig
-	if err := getCNIConfigOverrides(cfg, hc); err != nil {
-		return nil, err
+	c.CalicoCNICfg.CalicoConfig.KubeConfig, err = c.createCalicoKubeConfig(ctx, restConfig)
+	if err != nil {
+		return err
 	}
 
-	agent.NetworkName = CalicoHnsNetworkName
+	return nil
+}
 
-	if err := createNodeNameFile(cfg, dataDir); err != nil {
-		return nil, err
+// writeConfigFiles writes the three required files by Calico
+func (c *Calico) writeConfigFiles(CNIConfDir string, NodeName string) error {
+	// parse the two config files required by Calico
+	templates, err := parseCalicoTemplatesFromConfig(c.CalicoCNICfg)
+	if err != nil {
+		return err
+	}
+	
+	// create calico kubeconfig, calico config and node name files
+	if err = util.WriteFile(filepath.Join("c:\\", c.DataDir, "agent", CalicoKubeConfigName), templates[calicoKubeConfigTemplate]); err != nil {
+		return err
+	}
+	if err = util.WriteFile(filepath.Join(CNIConfDir, CalicoConfigName), templates[calicoConfigTemplate]); err != nil {
+		return err
+	}
+    if err = util.WriteFile(c.CalicoCNICfg.CalicoConfig.KubeConfig.Path, NodeName); err != nil {
+		return err
 	}
-	if err := createKubeConfig(cfg, dataDir); err != nil {
-		return nil, err
+
+	return nil
+}
+
+
+// createCalicoKubeConfig creates all needed for Calico to contact kube-api
+func (c *Calico) createCalicoKubeConfig(ctx context.Context, restConfig *rest.Config) (CalicoKubeConfig, error) {
```

Comment:
Should this perhaps return:
```suggestion
func (c *Calico) createCalicoKubeConfig(ctx context.Context, restConfig *rest.Config) (*CalicoKubeConfig, error) {
```

Reply:
Yes! Pointers improve efficiency